animations/tween-animations.md

```diff
@@ -0,0 +1,726 @@
+      [[highlight]]..addListener(() {[[/highlight]]
+        [[highlight]]setState(() {[[/highlight]]
+          [[highlight]]// the state that has changed here is the animation object’s value[[/highlight]]
+        [[highlight]]});[[/highlight]]
+      [[highlight]]});[[/highlight]]
+    [[highlight]]controller.forward();[[/highlight]]
+  [[highlight]]}[[/highlight]]
+
+  @override
+  Widget build(BuildContext context) {
+    return new Center(
+      child: new Container(
+        margin: new EdgeInsets.symmetric(vertical: 10.0),
+        height: [[highlight]]animation.value,[[/highlight]]
+        width: [[highlight]]animation.value,[[/highlight]]
+        child: new FlutterLogo,
+      ),
+    );
+  }
+}
+
+void main() {
+  runApp(new LogoApp());
+}
+{% endprettify %}
+
+The `addListener()` function calls `setState()`, so every time the
+Animation generates a new number, the current frame is marked dirty,
+which forces `build()` to be called again.
+In `build()`, the container changes size because its height and width
+now use `animation.value` instead of a hardcoded value.
+
+With these few changes, you’ve created your first animation in Flutter!
+You can find the source for this example,
+[animate1,](https://raw.githubusercontent.com/InMatrix/animation_tutorial/master/lib/animate1.dart)
+on GitHub.
+
+<aside class="alert alert-success" markdown="1">
+**Dart language tricks**
+You may not be familiar with Dart's cascade notation&mdash;the two
+dots in `..addListener()`. This syntax means that the `addListener()`
+method is called with the return value from `animate()`.
+Consider the following example:
+
+<!-- skip -->
+{% prettify dart %}
+[[highlight]]animation = tween.animate(controller)[[/highlight]]
+          [[highlight]]..addListener(()[[/highlight]] {
+            setState(() {
+              // the animation object’s value is the changed state
+            });
+          });
+{% endprettify %}
+
+This code is equivalent to:
+
+<!-- skip -->
+{% prettify dart %}
+[[highlight]]animation = tween.animate(controller);[[/highlight]]
+[[highlight]]animation.addListener(()[[/highlight]] {
+            setState(() {
+              // the animation object’s value is the changed state
+            });
+          });
+{% endprettify %}
+
+The difference is that, in the first case,
+the animation variable hasn't been set when the closure is created.
+Therefore, if you need to use `animation.value` in `setState()`,
+you must use the second form.
+
+You can learn more about cascade notation in the
+[Dart Language Tour.](https://www.dartlang.org/guides/language/language-tour)
+</aside>
+
+##  Simplifying with AnimatedWidget
+
+The call to `setState()` is an obvious candidate for refactoring,
+which you’ll find in a helper class named AnimatedWidget.
+The simplified code is shown below. LogoApp now derives from
+AnimatedWidget instead of from StatefulWidget, and your
+AnimatedWidget does not need to maintain a State object to
```

Comment:
But the parent of the AnimatedWidget needs to maintain a State object to hold the animation. I'm not sure if we should explain that.